AppDB/appscale/datastore/scripts/datastore.py

```diff
@@ -889,32 +900,49 @@ def main():
   server_node = '{}/{}:{}'.format(DATASTORE_SERVERS_NODE, options.private_ip,
                                   options.port)
 
+  retry_policy = KazooRetry(max_tries=5)
+  zk_client = kazoo.client.KazooClient(
+    hosts=zookeeper_locations, connection_retry=ZK_PERSISTENT_RECONNECTS,
+    command_retry=retry_policy)
+  zk_client.start()
+
   if args.type == 'cassandra':
     datastore_batch = DatastoreFactory.getDatastore(
       args.type, log_level=logger.getEffectiveLevel())
     zookeeper = zktransaction.ZKTransaction(
-      host=zookeeper_locations, db_access=datastore_batch,
+      zk_client=zk_client, db_access=datastore_batch,
       log_level=logger.getEffectiveLevel())
-    transaction_manager = TransactionManager(zookeeper.handle)
+    transaction_manager = TransactionManager(zk_client)
     datastore_access = DatastoreDistributed(
       datastore_batch, transaction_manager, zookeeper=zookeeper,
       log_level=logger.getEffectiveLevel(),
       taskqueue_locations=taskqueue_locations)
   else:
     from appscale.datastore.fdb.fdb_datastore import FDBDatastore
+    clusterfile_path = args.fdb_clusterfile
+    if not clusterfile_path:
+      try:
+        clusterfile_content = zk_client.get(FDB_CLUSTERFILE_NODE)[0]
+        clusterfile_path = '/tmp/appscale-datastore-fdb.cluster'
```

Comment:
/tmp may not be a good location, perhaps /run/appscale/... or /var/lib/appscale/...?

Reply:
Is there some guideline on how to use linux directories properly?

Reply:
In this case it seems like the file:

- is long lived
- has a well known name

which suggests `/tmp` is not the best location.

Reply:
(facepalm) I don't know what I thought about. Yes, I see that this case is not the one requiring a guideline to understand that it's not tmp file.
I'll change it tomorrow.